src/main/java/frc/robot/subsystems/vision/VisionIO.java

```diff
@@ -0,0 +1,19 @@
+package frc.robot.subsystems.vision;
+
+import java.util.Optional;
+
+import org.photonvision.EstimatedRobotPose;
+import org.photonvision.PhotonCamera;
+import org.photonvision.targeting.PhotonPipelineResult;
+
+import edu.wpi.first.math.geometry.Pose2d;
+
+public interface VisionIO {
+    public default PhotonCamera getCamera () { return new PhotonCamera(""); }
+
+    public default void update (Pose2d simRobotPoseMeters) {}
+
+    public default PhotonPipelineResult getResult () { return new PhotonPipelineResult(); };
+
+    public default Optional<EstimatedRobotPose> getEstimatedPosition () { return Optional.of(new EstimatedRobotPose(null, 0, null, null)); }
```

Comment:
Why do we want this to be an Optional?

Reply:
The photon pose estimator returns an optional, i left unwrapping it for the navigation method. But I could also easily unwrap it within the vision io method and just return null when there is no estimation. Would that be more concise?

Reply:
Got it, this seems fine. Since either way is acceptable we can leave it how you have it right now